src/pkg/images/unpack.go

```diff
@@ -0,0 +1,202 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package images provides functions for building and pushing images.
+package images
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/defenseunicorns/pkg/helpers/v2"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/zarf-dev/zarf/src/api/v1alpha1"
+	"github.com/zarf-dev/zarf/src/pkg/archive"
+	"github.com/zarf-dev/zarf/src/pkg/logger"
+	"github.com/zarf-dev/zarf/src/pkg/transform"
+	"github.com/zarf-dev/zarf/src/pkg/utils"
+	"oras.land/oras-go/v2"
+	"oras.land/oras-go/v2/content/oci"
+)
+
+// ImageWithManifest represents an image reference and its associated OCI manifest.
+type ImageWithManifest struct {
+	Image    transform.Image
+	Manifest ocispec.Manifest
+}
+
+const (
+	// This is the default docker annotation for the image name
+	dockerRefAnnotation = "io.containerd.image.name"
+	// When the Docker engine containerd image store is used only this annotation is used for sha referenced images
+	dockerContainerdImageStoreAnnotation = "containerd.io/distribution.source.docker.io"
+)
+
+// Unpack extracts an image tar and loads it into an OCI layout directory.
+// It returns a list of ImageWithManifest for all images in the tar.
+func Unpack(ctx context.Context, imageArchive v1alpha1.ImageArchive, destDir string, arch string) (_ []ImageWithManifest, err error) {
+	if imageArchive.Images == nil {
```

Comment:
Current conditional allows an empty list to proceed

```suggestion
	if len(imageArchive.Images) == 0 {
```